VideoLocker/src/main/java/org/edx/mobile/view/CourseBaseActivity.java

```diff
@@ -0,0 +1,140 @@
+package org.edx.mobile.view;
+
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.View;
+import android.widget.ProgressBar;
+
+import org.edx.mobile.R;
+import org.edx.mobile.base.BaseFragmentActivity;
+import org.edx.mobile.model.api.EnrolledCoursesResponse;
+import org.edx.mobile.util.AppConstants;
+import org.edx.mobile.util.NetworkUtil;
+import org.edx.mobile.view.custom.ETextView;
+
+/**
+ *  A base class to handle some common task
+ *  NOTE- in the layout file,  these should be defined
+ *  1. offlineBar
+ *  2. progress_spinner
+ *  3. offline_mode_message
+ */
+public abstract  class CourseBaseActivity  extends BaseFragmentActivity {
+
+    private View offlineBar;
+    private ProgressBar progressWheel;
+
+    protected Bundle bundle;
+    protected EnrolledCoursesResponse courseData;
+
+    @Override
+    protected void onCreate(Bundle arg0) {
+        super.onCreate(arg0);
+
+    }
+    protected void initialize(Bundle arg){
+        bundle = getIntent().getBundleExtra(Router.EXTRA_BUNDLE);
+        courseData = (EnrolledCoursesResponse) bundle
+            .getSerializable(Router.EXTRA_ENROLLMENT);
+
+        offlineBar = findViewById(R.id.offline_bar);
+        progressWheel = (ProgressBar) findViewById(R.id.progress_spinner);
+        if (!(NetworkUtil.isConnected(this))) {
+            AppConstants.offline_flag = true;
+            invalidateOptionsMenu();
+            showOfflineMessage();
+            if(offlineBar!=null){
+                offlineBar.setVisibility(View.VISIBLE);
+            }
+        }
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        setTitle(getString(R.string.find_courses_title));
```

Comment:
Since this an abstract base class maybe it doesn't need a default title? Seems like if it has one, we'll be more likely to leave the wrong one by accident?